### ISSUE TITLE
Best way to add environmental variable to set Fish configuration directory.

### ISSUE THREAD

**Issue by frederickjh** [xxh/xxh-shell-fish#2]
This is related to xxh/xxh-shell-fish#1 and this [comment](https://github.com/xxh/xxh-shell-fish/issues/1#issuecomment-604573610).

Fish normally finds or puts its configuration files in `~/.config/fish`. This can be changed by setting the environmental variable `XDG_CONFIG_HOME'. 

Using this line from `entrypoint.sh` as starting point,
```bash
export XXH_HOME=`dirname $CURRENT_DIR/../../../../p`
```
We could add something like this after it,
```bash
export XDG_CONFIG_HOME=`dirname $CURRENT_DIR/../../../../.xxh-shell-fish/config`
```
to set the configuration directory for the Fish shell.

**Comment by anki-code** [comment]
Common way is to place default config in the xxh-shell directory. 

Xonsh shell example:
1. [xonshrc.xsh is in the shell directory](https://github.com/xxh/xxh-shell-xonsh-appimage) - it will be [copied to build dir by build.sh](https://github.com/xxh/xxh-shell-xonsh-appimage/blob/master/build.sh#L20) and will be in `~/.xxh/xxh/shells/xxh-shell-xonsh-appimage/build` directory along with `entrypoint.sh`
2. when you do `xxh myhost` it upload build dir to the host and calls [entrypoint.sh which calls xonshrc.xsh file when run xonsh](https://github.com/xxh/xxh-shell-xonsh-appimage/blob/master/entrypoint.sh#L80)
3. [xonshrc.xsh init/source/import plugins](https://github.com/xxh/xxh-shell-xonsh-appimage/blob/master/xonshrc.xsh#L51-L60)

On the step 2 and 3 inside `shellrc` and `pluginrc` scripts you can get the directory where this scripts are placed (or use global environment variable [$XXH_HOME which set in entrypoint.sh](https://github.com/xxh/xxh-shell-fish/blob/master/entrypoint.sh#L78)). 

If you need to copy config files from xxh-shell to $XXH_HOME to save this between xxh updates:
1. Add this files to build dir by modifying `xxh-shell-fish/build.sh`
2. Add the code to `fishrc.fish` which check that `$XXH_HOME/.fish-configs` is not exists and make copy from `~/.xxh/xxh/shells/xxh-shell-fish/build/fish-configs` to `$XXH_HOME/.fish-configs`.
3. If `$XXH_HOME/.fish-configs` exists then load configs from there.

But in common case it's not needed. Configs could be a part of the `xxh-shell-fish/build` directory and in bash, zsh, xonsh I avoided using `$XXH_HOME`. My be my cases were simple.

**Comment by frederickjh** [comment]
Fish shell documentation on [Initialization files](https://fishshell.com/docs/current/index.html#initialization) and [Startup](https://fishshell.com/docs/current/tutorial.html#startup-where-s-bashrc).

**Comment by frederickjh** [comment]
Closing as it was fixed in commit 5a3181b.
